forte/data/base_pack.py

```diff
@@ -550,11 +566,24 @@ def entry_setter(cls: Entry, value: Any, attr_name: str, field_type):
                 tid=cls.tid, attr_name=attr_name, attr_value=attr_value
             )
 
+        # If this is the first time an entry of this type is
+        # created, its attributes do not have a getter and setter
+        # property associated with them. In this case, we manually
+        # add all dataclass attributes to the cached_attribute_data
+        # dict.
+        for name in entry.__dataclass_fields__:
```

Comment:
This is correct. However, the first two points are relevant for initialization and the third point is useful otherwise. I will make sure to specify all these scenarios in required places.